### PR TITLE
internet_explorer

### PR DESCRIPTION
**Pull request by Tobias-B-Besemer**
Try to fix issue #460.

### CODE REVIEW

File: cleaners/internet_explorer.xml [hunk 1]
```diff
@@ -40,20 +50,92 @@
...
+    <action command="winreg" path="HKCR\Local Settings\Software\Microsoft\Windows\CurrentVersion\AppContainer\Storage\windows_ie_ac_001\Internet Explorer\DOMStorage"/>
+    <action command="winreg" path="HKCU\Software\Microsoft\Internet Explorer\LowRegistry\DOMStorage"/>
+    <!-- Used e.g. in Windows XP (and maybe before on NT) - English-only-Support: -->
+    <!-- Was, and didn't worked: -->
+    <!-- <action command="delete" search="walk.files" path="%USERPROFILE%\Local Settings\Temporary Internet Files\"/> -->
+    <action command="delete" search="walk.all" path="%USERPROFILE%\Local Settings\Temporary Internet Files\"/>
+    <action command="delete" search="glob" path="%USERPROFILE%\Local Settings\Temporary Internet Files\*"/>
+    <!-- Used e.g. in Windows XP (and maybe before on NT) - some none-English-Support: -->
+    <!-- Polish Windows XP Support: -->
+    <action command="delete" search="walk.all" path="%USERPROFILE%\Ustawienia lokalne\Temporary Internet Files\"/>
```
Comment (az0) [inline cleaners/internet_explorer.xml]: I could merge the Polish parts for BleachBit version 2.1/2.2 if it were a separate pull request (it's up to you). This is a small change, and the bug reporter confirmed the fix.

There are some big changes in the whole pull request to an important application (MSIE), so the rest will need to wait for version 2.3/2.4. Thanks for your patience

Comment (Tobias-B-Besemer) [reply cleaners/internet_explorer.xml]: Here is a PR for Polish Windows XP Support: https://github.com/bleachbit/bleachbit/pull/485

Comment (Tobias-B-Besemer) [reply cleaners/internet_explorer.xml]: Maybe you should also merge https://github.com/bleachbit/bleachbit/pull/475 for BB2.2... it's also Polish support... It got confirmed here: https://github.com/bleachbit/bleachbit/issues/460#issuecomment-471987298

**Comment by az0** [comment]
I set milestone for 2.4 to allow for checking all the many changes

File: cleaners/internet_explorer.xml [hunk 2]
```diff
@@ -12,34 +12,43 @@
...
   <option id="cookies">
     <label>Cookies</label>
     <description>Delete cookies, which contain information such as web site preferences, authentication, and tracking identification</description>
-    <action command="delete" search="walk.files" path="%USERPROFILE%\Cookies\"/>
-    <action command="delete" search="walk.files" path="%APPDATA%\Microsoft\Windows\Cookies\"/>
+    <action command="delete" search="walk.files" path="%UserProfile%\Cookies\"/>
+    <action command="delete" search="walk.files" path="%AppData%\Microsoft\Windows\Cookies\"/>
     <action command="delete" search="walk.files" path="%LocalAppData%\Microsoft\Internet Explorer\DOMStore\"/>
-    <action command="delete" search="walk.files" path="%LocalAppData%\Microsoft\Windows\INetCookies\"/>
+    <action command="delete" search="walk.files" path="$IELocalAppData\INetCookies\"/>
```
Comment (az0) [inline cleaners/internet_explorer.xml]: Line 51 should be `$$IELocalAppData$$` instead of `$IeLocalAppData`?

Comment (Tobias-B-Besemer) [reply cleaners/internet_explorer.xml]: Yes, sorry! I fix that!

**Comment by az0** [comment]
My notes

**Change log**
Change casing of variables from uppercase to camel case
Use more specific variables that refer to deeper paths
Clean more cookies
Refactor for multi-value variables
Clean more registry keys and files for history
Clean cache (new option)
Clean downloads (new option)

**To-do**
Second wildcard on CryptnetUrlCache looks unnecessary
WebCache.old search=file can be changed to search=top to eliminate one action
New option "last download location" can be moved under history option

**Comment by Tobias-B-Besemer** [comment]
Andrew, I will try to spend more time again for BB and get rid of it! Hope I still have time till release of v3...

**Comment by Tobias-B-Besemer** [comment]
Btw.: I never clean download history, so I'm not sure about combine it...

**Comment by Tobias-B-Besemer** [comment]
<3 ^^

**Pull request merged**